### PR TITLE
fix: config file non-fatal edge cases, tests

### PR DESCRIPTION
**Pull request by avirtopeanu-ionos**
## What does this fix or implement?

certain edge cases for reading config file would fatally stop the execution when instead should recover

## Checklist

- [x] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [x] Tests added or updated
- [ ] Documentation updated
- [ ] Sonar Cloud Scan
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

**Review by gabisavu-ionos — APPROVED** [review]

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -12,19 +12,156 @@ import (
...
+func TestTokEnvWithFullConfig(t *testing.T) {
+	// Config token should not override env var token
+	os.Clearenv()
+	viper.Reset()
+
+	assert.NoError(t, os.Setenv(sdk.IonosTokenEnvVar, "tok"))
+	path := filepath.Join("..", "testdata", "config.json") // TODO: These files should be created and then destroyed by the tests
+	viper.SetConfigFile(path)
+	viper.Set(constants.ArgConfig, path)
+	assert.NoError(t, os.Chmod(path, 0600))
```
Comment (adeatcu-ionos) [inline pkg/config/config_test.go:57]: @avirtopeanu-ionos not a big deal, you can leave it as it is, but since now you are only reading from the file, the number can be `0400`. You can leave it as it is since in the future you will create the test file, write some info in it, use the test file for the test and then delete the test file.

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -12,19 +12,156 @@ import (
...
+	assert.Equal(t, "", viper.GetString(ServerUrl))
+}
+
+func TestTokEnvWithUserPassConfigBackup(t *testing.T) {
+	// Useful for API routes which don't accept bearer tokens, or custom IonosCTL commands (Image Upload)
+	os.Clearenv()
+	viper.Reset()
+
+	assert.NoError(t, os.Setenv(sdk.IonosTokenEnvVar, "tok"))
+	path := filepath.Join("..", "testdata", "config_user_pass.json") // TODO: These files should be created and then destroyed by the tests
```
Comment (adeatcu-ionos) [inline pkg/config/config_test.go:36]: @avirtopeanu-ionos good mention for `TODO`.

File: pkg/config/config_test.go (hunk 1 above)
Comment (avirtopeanu-ionos) [reply pkg/config/config_test.go:57]: there had been a discussion previouslyabout the config file permissions. The idea is that LoadFile() fails if the perms aren't exactly `0600`. I can't comment on why this is exactly, but this is the reason why we set exactly this perm number

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -12,19 +12,156 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestUsingJustTokenEnvVar(t *testing.T) {
+	os.Clearenv()
```
[the remark this answers is not in the document]
Comment (avirtopeanu-ionos) [reply pkg/config/config_test.go:16]: good call. On my end these foobar env vars dont carry over. Checking official go tests there doesn't seem to be any special "cleaning procedure" in place https://github.com/golang/go/blob/6fd0520db3cffad4fd01c98319b9b1bf3c50f5e2/src/cmd/go/go_test.go#L280

[the remark this answers is not in the document]
Comment (avirtopeanu-ionos) [reply pkg/config/config_test.go:16]: https://unix.stackexchange.com/questions/372627/is-it-possible-to-pass-environment-variables-from-child-to-parent-in-user-space

**Review by mimihalescu — APPROVED** [review]

**Comment by avorima** [comment]
Can you make a release ASAP?
